feature/gribi/basic_encap/README.md

```diff
@@ -0,0 +1,389 @@
+                            ipv6 {
+                                dscp-set: [dscp_encap_b_1, dscp_encap_b_2]
+                            }
+                            action {
+                                network-instance: "ENCAP_TE_VRF_B"
+                            }
+                        }
+                        rule {
+                            sequence-id: 17
+                            action {
+                                network-instance: "DEFAULT"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+*   Using gRIBI, install the following gRIBI AFTs, and validate the specified
+    behavior.
+
+```
+IPv6Entry {2015:aa8::/32 (ENCAP_TE_VRF_A)} -> NHG#10 (DEFAULT VRF)
+IPv4Entry {138.0.11.0/24 (ENCAP_TE_VRF_A)} -> NHG#10 (DEFAULT VRF) -> {
+  {NH#201, DEFAULT VRF, weight:1},
+  {NH#202, DEFAULT VRF, weight:3},
+}
+NH#201 -> {
+  encapsulate_header: OPENCONFIGAFTTYPESENCAPSULATIONHEADERTYPE_IPV4
+  ip_in_ip {
+    dst_ip: "203.0.113.1"
+    src_ip: "ipv4_outer_src_111"
+  }
+  network_instance: "TE_VRF_111"
+}
+NH#202 -> {
+  encapsulate_header: OPENCONFIGAFTTYPESENCAPSULATIONHEADERTYPE_IPV4
+  ip_in_ip {
+    dst_ip: "203.10.113.2"
+    src_ip: "ipv4_outer_src_111"
+  }
+  network_instance: "TE_VRF_111"
+}
+
+// 203.0.113.1 is the tunnel IP address.
+
+IPv4Entry {203.0.113.1/32 (TE_VRF_111)} -> NHG#1 (DEFAULT VRF) -> {
+  {NH#1, DEFAULT VRF, weight:1,ip_address=192.0.2.111},
+  {NH#2, DEFAULT VRF, weight:3,ip_address=192.0.2.222},
+}
+IPv4Entry {192.0.2.111/32 (DEFAULT VRF)} -> NHG#2 (DEFAULT VRF) -> {
+  {NH#10, DEFAULT VRF, weight:1,mac_address:magic_mac, interface-ref:dut-port-2-interface},
+  {NH#11, DEFAULT VRF, weight:3,mac_address:magic_mac, interface-ref:dut-port-3-interface},
+}
+IPv4Entry {192.0.2.222/32 (DEFAUlT VRF)} -> NHG#3 (DEFAULT VRF) -> {
+  {NH#100, DEFAULT VRF, weight:2,mac_address:magic_mac, interface-ref:dut-port-4-interface},
+  {NH#101, DEFAULT VRF, weight:3,mac_address:magic_mac, interface-ref:dut-port-5-interface},
+}
+
+// 203.10.113.2 is the tunnel IP address. Note that the NHG#1 is shared by both tunnels.
+
+IPv4Entry {203.10.113.2/32 (TE_VRF_111)} -> NHG#1 (DEFAULT VRF) -> <omitted for brevity>
+```
+
+## Procedure
+
+#### Test-1, IPv4 traffic WCMP Encap
+
+Send packets to DUT port-1. The outer v4 header has the destination addresses
+138.0.11.8. Validate that:
+
+*   All egress packets (100%) are IPinIP (4in4) encapped.
+*   Packets are encapped to the tunnel IPs in the specified ratio. Specifically,
+    25% of the egress packets should have the destination address 203.0.113.1,
+    and 75% of the egress packets should have the destination address
+    203.10.113.2.
+*   The encapped/tunneled packets should be distributed hierarchically per the
+    weight.
+*   The DSCP value is copied from the inner header to the outer header.
+*   The TTL value is copied from the inner header to the outer header.
```

Comment:
Can you confirm if this is an expected requirement. As per my understanding the inner TTL value should not be copied to outer header.

Reply:
Yes. It's expected. Feel free to open a internal bug if you have further concerns. Thanks.